USERMANUAL.md

```diff
@@ -0,0 +1,596 @@
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - EQUAL_COLUMNS:
+          ref_table_id: <reference_table_id>
+          ref_column_id: <reference_column_id>
+```
+
+Full example: [Rule](docs/examples/row_by_row.yaml) and [rule binding](docs/examples/row_by_row.yaml)
+
+
+#### Equality of two columns (special case: columns do not contain repeated values)
+
+For this test we compare two columns that we know have distinct values (for example because we have checked this separately).
+
+```yaml
+rules:
+  EQUAL_COLUMNS_UNIQUE_VALUES:
+    rule_type: CUSTOM_SQL_STATEMENT
+    custom_sql_arguments:
+        - ref_table_id
+        - ref_column_id
+    params:
+      custom_sql_statement: |-
+        with t1 as (
+          select 
+            $column as t1_col, 
+            row_number() over (order by $column) as n 
+          from data),
+        t2 as (
+          select 
+            $ref_column_id as t2_col, 
+            row_number() over (order by $ref_column_id) as n 
+            from `<dataset_id>.$ref_table_id`
+        )
+        select t1.t1_col, t2.t2_col
+        from t1 
+        full outer join t2 using(n)
+        where not t1_col = t2_col 
+
+rule_bindings:
+  EQUAL_COLUMNS:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - EQUAL_COLUMNS_UNIQUE_VALUES:
+          ref_table_id: <reference_table_id>
+          ref_column_id: <reference_column_id>
+```
+
+Full example: [Rule](docs/examples/row_by_row.yaml) and [rule binding](docs/examples/row_by_row.yaml)
+
+
+#### Equality of an aggregate
+
+In this example we check that the aggregate in one table equals a value in another table. For instance, the sum over line items in an order should be equal to the total invoice amount in another table.
+
+This example uses a rule of type CUSTOM_SQL_STATEMENT with 5 parameters. We assume there are two tables, one where we will aggregated based on some key, and sum a value column, and another table that has the totals for each key. The parameters are 
+1. the column ID for the key and 
+1. the value that we are aggregated, 
+1. the column ID for the key and 
+1. the value in table that contains the totals, and 
+1. the table ID of the table containing the totals.
+
+As we specified everything that we need through the above “custom SQL arguments”, the column_id field in the rule binding has no effect. 
+
+```yaml
+rules:
+  EQUAL_AGGREGATE:
+    rule_type: CUSTOM_SQL_STATEMENT
+    custom_sql_arguments: 
+      key_column_id
+      value_column_id
+      ref_table_id
+      ref_key_column_id
+      ref_value_column_id
+    params:
+      custom_sql_statement: |-
+        with t1 as (
+          select $key_column_id as key, sum($value_column_id) as total
+          from data
+          group by $key_column_id),
+        t2 as (
+          select $ref_key_column_id as key, sum($ref_value_column_id) as ref_total
+          from <dataset_id>.$ref_table_id
```

Comment:
I suggest escaping all table references, e.g.  s/<dataset_id>.$ref_table_id<dataset_id>.$ref_table_id/`$reg_table_id`/

Reply:
Like this, right?
``s/<dataset_id>.$ref_table_id/<dataset_id>.`$ref_table_id`/``

Reply:
Better escape the whole dataset ID as well, no?
``s/<dataset_id>.$ref_table_id/`<dataset_id>.$ref_table_id`/``